src/Multi_filtration/include/gudhi/One_critical_filtration.h

```diff
@@ -39,266 +42,477 @@ namespace Gudhi::multi_filtration {
  * x>=(2,1)\} \f$ is finitely critical, and more particularly 2-critical, while
  *  - \f$ \{ x \in  \mathbb R^2 : x>= \mathrm{epigraph}(y\mapsto e^{-y})\} \f$
  * is not.
- * Use \ref Multi_critical_filtration for multicritical filtrations.
- * \tparam T value type of the vector-like.
+ * Use \ref Multi_critical_filtration for multi-critical filtrations.
+ * \tparam T value type of the vector-like. Has to implement std::isnan(), std::numeric_limits<T>::has_quiet_NaN,
+ * std::numeric_limits<T>::quiet_NaN(), std::numeric_limits<T>::has_infinity, std::numeric_limits<T>::infinity(),
+ * std::numeric_limits<T>::max().
  */
 template <typename T>
-class One_critical_filtration : public std::vector<T> {
+class One_critical_filtration : public std::vector<T>
+{
+ private:
+  using Base = std::vector<T>;
+
  public:
-  One_critical_filtration() : std::vector<T>() {};
-  One_critical_filtration(int n) : std::vector<T>(n, -T_inf) {};  // minus infinity by default
-  One_critical_filtration(int n, T value) : std::vector<T>(n, value) {};
-  One_critical_filtration(std::initializer_list<T> init) : std::vector<T>(init) {};
-  One_critical_filtration(const std::vector<T> &v) : std::vector<T>(v) {};
-  One_critical_filtration(std::vector<T> &&v) : std::vector<T>(std::move(v)) {};
+  template <typename U = T>
+  using Single_point = One_critical_filtration<U>;
+
+  // CONSTRUCTORS
+
+  One_critical_filtration() : Base() {};
+  // warning: can be problematic if the user never updates the values and let it like that, {-inf, -inf, ...} is not
+  // considered as -inf.
+  One_critical_filtration(int n) : Base(n, -T_inf) {};  // minus infinity by default
+  One_critical_filtration(int n, T value) : Base(n, value) {};
+  One_critical_filtration(std::initializer_list<T> init) : Base(init) {};
+  One_critical_filtration(const std::vector<T> &v) : Base(v) {};
+  One_critical_filtration(std::vector<T> &&v) : Base(std::move(v)) {};
   One_critical_filtration(typename std::vector<T>::iterator it_begin, typename std::vector<T>::iterator it_end)
-      : std::vector<T>(it_begin, it_end) {};
+      : Base(it_begin, it_end) {};
   One_critical_filtration(typename std::vector<T>::const_iterator it_begin,
                           typename std::vector<T>::const_iterator it_end)
-      : std::vector<T>(it_begin, it_end) {};
+      : Base(it_begin, it_end) {};
+
+  One_critical_filtration &operator=(const One_critical_filtration &a)
+  {
+    Base::operator=(a);
+    return *this;
+  }
+
+  // HERITAGE
 
   using std::vector<T>::operator[];
   using value_type = T;
-  using OneCritical = One_critical_filtration<T>;
-  template <typename value_type>
-  using Base = One_critical_filtration<value_type>;
+
+  // CONVERTERS
+
   operator std::vector<T> &() const { return *this; }
-  std::vector<T> get_vector() const { return static_cast<std::vector<T>>(*this); }
-  std::size_t num_parameters() const { return this->size(); }
```

Comment:
I'm in favor of removing the `size` method, in favor of the `num_parameters` here,
as in the multicritical case, it can be confusing -> so `num_generators`  and `num_parameters` are easier to understand. 
If we want to stay coherent with the methods name, I think this should be better.

Edit: ok its moved down